### DISCUSSION TITLE
How to add a folder and its contents to windowed bundle?

### DISCUSSION THREAD

**Discussion by Moosems** [pyinstaller/pyinstaller#6933] [Help] [answered]
How would I add a folder and its contents to windowed bundle from the spec file?

**Comment by rokm** [comment, marked as answer]
See https://pyinstaller.org/en/stable/spec-files.html#adding-data-files

You pass the `datas=[('path/to/src/dirname', 'dirname')]` to `Analysis`, and the source directory `dirname` ends up collected as target directory `dirname` inside `name.app/Contents/Resources` and symlinked into `name.app/Contents/MacOS`. (Well, unless you're making a onefile app bundle, in which case the data directory is collected into executable-embedded archive and extracted to temporary directory along with other contents when you run the application).

**Reply by Moosems** [reply]
For some reason if I include a folder it makes the window look blurry

**Reply by Moosems** [reply]
With:
<img width="312" alt="Screen Shot 2022-07-09 at 11 47 57 AM" src="https://user-images.githubusercontent.com/95927277/178115204-de8052b3-fdc6-4379-b831-c6e2f304b802.png">

Without:
<img width="312" alt="Screen Shot 2022-07-09 at 11 51 52 AM" src="https://user-images.githubusercontent.com/95927277/178115354-64a58090-0cde-4acc-b48b-bdf3cae5fa8e.png">

**Reply by rokm** [reply]
What is the content of the folder?

**Reply by Moosems** [reply]
A png file and an icns file
